File: NetFramework/XHtmlEditor/CS/RichTextEditor/BackstageTabs/OpenTab.cs
using RichTextEditor.BackstageTabs.Items;
using System;
using System.Drawing;

namespace RichTextEditor.BackstageTabs
{
    public partial class OpenTab : BackstageTabFileList
    {
        public OpenTab()
        {
            Button.ItemImage = Properties.Resources.Open_large;
            Button.ItemText = Properties.Resources.Browse_txt;
            Caption = GetElementText(Properties.Resources.OpenTab_txt);
        }

        #region override
        protected override void OnMenuItemClick(BackstageTabItem mi)
        {
            var ribbon = (C1TextEditorRibbon)BackstageView.Ribbon;
            ribbon.OpenDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text));
            mi.Item.Date = DateTime.Now;
        }

        protected override void OnMenuButtonClick()
        {
            var ribbon = (C1TextEditorRibbon)BackstageView.Ribbon;
            ribbon.OpenDocument();
        }
        #endregion        
    }
}

File: NetFramework/XHtmlEditor/VB/RichTextEditor/BackstageTabs/OpenTab.vb
﻿
Imports RichTextEditor.RichTextEditor

Partial Public Class OpenTab
    Inherits BackstageTabFileList

    Public Sub New()
        Button.ItemImage = Resources.Open_large
        Button.ItemText = Resources.Browse_txt
        Caption = GetElementText(Resources.OpenTab_txt)
    End Sub

    Protected Overrides Sub OnMenuItemClick(ByVal mi As BackstageTabItem)
        Dim ribbon As C1TextEditorRibbon = CType(BackstageView.Ribbon, C1TextEditorRibbon)
        ribbon.OpenDocument(System.IO.Path.Combine(mi.Item.SubText, mi.Item.Text))
        mi.Item.ItemDate = DateTime.Now
    End Sub

    Protected Overrides Sub OnMenuButtonClick()
        Dim ribbon As C1TextEditorRibbon = CType(BackstageView.Ribbon, C1TextEditorRibbon)
        ribbon.OpenDocument()
    End Sub
End Class